test/blackbox/common/DDSBlackboxTestsOwnershipQos.cpp

```diff
@@ -2318,12 +2275,32 @@ TEST(OwnershipQos, exclusive_kind_keep_all_reliable_mixed)
     drop_messages.store(false);
     EXPECT_TRUE(high_strength_writer.waitForAllAcked(std::chrono::seconds(1)));
 
-    // Make the reader process the data, expecting only the high strength data.
+    // Make the reader process the data, expecting only the required data.
     // The issue was reproduced by the reader complaining about reception of unexpected data.
     reader.startReception(expected_data);
     reader.block_for_all();
 }
```

Comment:
Should we check the expected amount of samples received too?

Reply:
We are not doing such checks in other tests using `block_for_all`.
I guess it is implicitly checking the number of received samples.